_posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md

```diff
@@ -0,0 +1,151 @@
+
+```
+application-log-pipeline:
+  workers: 4
+  delay: "50"
+  source:
+    http:
+  processor:
+    - grok:
+        match:
+          log: [ "%{NOTSPACE:time} %{NOTSPACE:thread} %{NOTSPACE:loglevel}  %{NOTSPACE:class} - %{GREEDYDATA:message}" ]
+
+  route:
+    - warn_and_above: '/loglevel == "WARN" or /loglevel == "ERROR"'
+  sink:
+    - opensearch:
+        routes:
+          - warn_and_above
+        hosts: ["https://opensearch:9200"]
+        insecure: true
+        username: "admin"
+        password: "admin"
+        index: warn-and-above-logs
+    - opensearch:
+        hosts: ["https://opensearch:9200"]
+        insecure: true
+        username: "admin"
+        password: "admin"
+        index: all-logs
+```
+
+There are many other use-cases that conditional routing can support. If there are other conditional expressions 
+you’d like to see support for, please create an issue in GitHub.
+
+## Peer Forwarder
+
+Data Prepper supports stateful aggregations for traces and logs. With these, pipeline authors can improve the quality of the data going into OpenSearch. 
+Previous to Data Prepper 2.0, performing stateful trace aggregations required using the `peer-forwarder` processor plugin. 
+But this plugin only worked for traces and would send data back to the source. Also, log aggregations only worked on a single node.
+
+Data Prepper introduces peer forwarding as a core feature in Data Prepper 2.0. This allows pipeline authors to perform stateful 
+aggregations on multiple Data Prepper nodes. When performing stateful aggregations, Data Prepper uses a hash ring to determine 
+which nodes are responsible for processing different events based on the values of certain fields. Data Prepper's core peer-forwarder 
+routes events to the node responsible for processing the event. That node then holds all the state necessary for performing the aggregation.
+
+To use peer forwarding, you will configure how Data Prepper discovers other nodes and the security for connections in your 
+`data-prepper-config.yaml` file. The following snippet shows an example of how to do this.
+
+```
+peer_forwarder:
+    discovery_mode: dns
+    domain_name: "my-data-prepper-cluster.production"
+    ssl_certificate_file: /usr/share/data-prepper/config/my-certificate.crt
+    ssl_key_file: /usr/share/data-prepper/config/my-certificate.key
+    ssl_fingerprint_verification_only: true
+    authentication:
+        mutual_tls:
+```
+
+In the example above, Data Prepper will discover other peers using DNS. It will perform a DNS query on the domain `my-data-prepper-cluster.production`.
+This DNS record should be an A record with a list of IP addresses for peers. The configuration uses a custom certificate and private key. 
+It performs host verification by checking the fingerprint of the certificate. And finally it configures each server to authenticate requests using 
+Mutual TLS (mTLS) to prevent tampering of data.
+
+## Directory structure
```

Comment:
Suggested rewrites for directory structure section:

--- 

Before the release of Data Prepper 2.0, we distributed Data Prepper as a single executable JAR file. While convenient, it made it difficult for us to include custom plugins.

We now distribute Data Prepper 2.0 in a bundled directory structure. This structure features a shell script to launch Data Prepper and dedicated subdirectories for JAR files, configurations, pipelines, logs, and more.

```
data-prepper-2.0.0/
  bin/
    data-prepper                    # Shell script to run Data Prepper
  config/
    data-prepper-config.yaml        # The Data Prepper configuration file
    log4j.properties                # Logging configuration
  pipelines/                             # New directory for pipelines
    trace-analytics.yaml
    log-ingest.yaml
  lib/
    data-prepper-core.jar
    ... any other jar files
  logs/
```
You can launch Data Prepper by running `bin/data-prepper`; no need for additional command line arguments or Java system property definitions. Instead, the application loads from the `config/` subdirectory.

Data Prepper 2.0 reads pipeline configurations from the `pipelines/` subdirectory. You can now define pipelines across multiple YAML files in the subdirectory, where each file contains the settings for one or more pipelines. The directory also helps keep pipeline definition distinct and, therefore, more compact and focused.

Reply:
I added "You can _now_ launch Data Prepper..." to clarify this is new.

Also, "application loads _configurations_ from"

And I used definition here: "contains the _definition_ for one or more pipelines." I think settings is not quite clear.